src/Rings/mpoly-ideals.jl

```diff
@@ -2009,57 +2009,85 @@ function Base.show(io::IO, I::MPolyIdeal)
+the flag Plücker ideal with respect to the degree reverse lexicographical order.
+
 # Examples
 
 Complete flag variety $\text{Fl}(\mathbb{Q}, (1,2,3), 4)$. 
 ```jldoctest
 julia> flag_pluecker_ideal(QQ,[1,2,3],4)
 Ideal generated by
-  x[[1]]*x[[3, 4]] - x[[3]]*x[[1, 4]] + x[[4]]*x[[1, 3]]
-  -x[[1, 4]]*x[[2, 3]] + x[[2, 4]]*x[[1, 3]] - x[[3, 4]]*x[[1, 2]]
-  x[[2]]*x[[3, 4]] - x[[3]]*x[[2, 4]] + x[[4]]*x[[2, 3]]
-  x[[1]]*x[[2, 4]] - x[[2]]*x[[1, 4]] + x[[4]]*x[[1, 2]]
-  x[[1]]*x[[2, 3]] - x[[2]]*x[[1, 3]] + x[[3]]*x[[1, 2]]
-  x[[1]]*x[[2, 3, 4]] - x[[2]]*x[[1, 3, 4]] + x[[3]]*x[[1, 2, 4]] - x[[4]]*x[[1, 2, 3]]
-  -x[[1, 4]]*x[[1, 2, 3]] + x[[1, 3]]*x[[1, 2, 4]] - x[[1, 2]]*x[[1, 3, 4]]
-  -x[[2, 4]]*x[[1, 2, 3]] + x[[2, 3]]*x[[1, 2, 4]] - x[[1, 2]]*x[[2, 3, 4]]
+  x[[2, 4]]*x[[1, 2, 3]] - x[[2, 3]]*x[[1, 2, 4]] + x[[1, 2]]*x[[2, 3, 4]]
+  x[[1, 4]]*x[[1, 2, 3]] - x[[1, 3]]*x[[1, 2, 4]] + x[[1, 2]]*x[[1, 3, 4]]
   -x[[3, 4]]*x[[1, 2, 3]] - x[[1, 3]]*x[[2, 3, 4]] + x[[2, 3]]*x[[1, 3, 4]]
   -x[[1, 4]]*x[[2, 3, 4]] + x[[2, 4]]*x[[1, 3, 4]] - x[[3, 4]]*x[[1, 2, 4]]
+  -x[[1]]*x[[2, 3, 4]] + x[[2]]*x[[1, 3, 4]] - x[[3]]*x[[1, 2, 4]] + x[[4]]*x[[1, 2, 3]]
+  -x[[1, 4]]*x[[2, 3]] + x[[2, 4]]*x[[1, 3]] - x[[3, 4]]*x[[1, 2]]
+  -x[[1]]*x[[2, 3]] + x[[2]]*x[[1, 3]] - x[[3]]*x[[1, 2]]
+  -x[[2]]*x[[3, 4]] + x[[3]]*x[[2, 4]] - x[[4]]*x[[2, 3]]
+  -x[[1]]*x[[3, 4]] + x[[3]]*x[[1, 4]] - x[[4]]*x[[1, 3]]
+  -x[[1]]*x[[2, 4]] + x[[2]]*x[[1, 4]] - x[[4]]*x[[1, 2]]
 
 ```
 Flag variety $\text{Fl}(\mathbb{Q},(1,3),4)$.
 ```jldoctest
 julia> flag_pluecker_ideal(QQ,[1,3],4)
 Ideal generated by
-  x[[1]]*x[[2, 3, 4]] - x[[2]]*x[[1, 3, 4]] + x[[3]]*x[[1, 2, 4]] - x[[4]]*x[[1, 2, 3]]
+  -x[[1]]*x[[2, 3, 4]] + x[[2]]*x[[1, 3, 4]] - x[[3]]*x[[1, 2, 4]] + x[[4]]*x[[1, 2, 3]]
+```
+An example with a custom ring as input.
+```jldoctest
+julia> R, _ = polynomial_ring(QQ, 8)
+(Multivariate polynomial ring in 8 variables over QQ, QQMPolyRingElem[x1, x2, x3, x4, x5, x6, x7, x8])
+
+julia> flag_pluecker_ideal(R, [1,3], 4; minimal=false)
+Ideal generated by
+  -x1*x6 + x2*x5 - x3*x7 + x4*x8
 ```
 """
-function flag_pluecker_ideal(F::Field, dimensions::Vector{Int}, n::Int)
+function flag_pluecker_ideal(F::Field, dimensions::Vector{Int}, n::Int; minimal::Bool=true)
+  L = sum(binomial(n, d) for d in dimensions)
+  ring, _ = polynomial_ring(F, L; cached=false)
+
+  return flag_pluecker_ideal(ring, dimensions, n; minimal=minimal)
+
+end
+
+flag_pluecker_ideal(dimensions::Vector{Int}, n::Int; minimal::Bool=true) = flag_pluecker_ideal(QQ,dimensions,n, minimal=minimal)
+
+function flag_pluecker_ideal(ring::MPolyRing{<: FieldElem}, dimensions::Vector{Int}, n::Int; minimal::Bool=true)
   dimensions = unique(dimensions)
+   
+  @req maximum(dimensions) <= n-1 "The dimensions must be at most n-1"
+  @req sum(binomial(n, d) for d in dimensions) == ngens(ring) "The ring doesn't have the right number of variables"
+  @req dimensions == sort(dimensions) "The dimensions must be increasing"
```

Comment:
```suggestion
  @req issorted(dimensions) "The dimensions must be increasing"
```
Checking is O(n)  while actually sorting O(n*log(n))